File: 169-majority-element/majority-element.ts
function majorityElement(nums: number[]): number {
    let countMap = new Map()
    const majorityThreshold: number = Math.ceil(nums.length/2)


    for(let num of nums){
        countMap.has(num) ? countMap.set(num, countMap.get(num) + 1) : countMap.set(num, 1)
    }
    for(let [key, value] of countMap){
        if(value >= majorityThreshold){
            return key
        }
    }
    console.log("MAP", countMap)



    return 0
    
};

File: 169-majority-element/majority-element.py
from collections import Counter
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        m = defaultdict(int)
        
        for num in nums:
            m[num] += 1
        
        n = len(nums) // 2
        for key, value in m.items():
            if value > n:
                return key
        
        return 0

        # nums.sort()
        # n = len(nums)
        # return nums[n//2]